head/src/main/kotlin/io/qalipsis/core/head/campaign/ChannelNameFactory.kt

```diff
@@ -0,0 +1,36 @@
+/*
+ * QALIPSIS
+ * Copyright (C) 2022 AERIS IT Solutions GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package io.qalipsis.core.head.campaign
+
+import io.qalipsis.api.context.NodeId
+import io.qalipsis.core.annotations.LogInput
+import io.qalipsis.core.campaigns.RunningCampaign
+
+internal interface ChannelNameFactory {
```

Comment:
Do not make it internal, otherwise you will not be able to implement it differently in the enteprise edition.